backend/app/services/diagnosis_agent.py

```
@@ -33,205 +33,529 @@ class DiagnosisState(TypedDict):
-                if cat == 'POP' and 'precipitation_prob' not in weather_dict: weather_dict['precipitation_prob'] = val
-                if cat == 'REH' and 'humidity' not in weather_dict: weather_dict['humidity'] = val
-                if cat == 'WSD' and 'wind_speed' not in weather_dict: weather_dict['wind_speed'] = val
-            
-            w_str = f"{weather_dict.get('temperature', '-')}℃, {weather_dict.get('humidity', '-')}%, {weather_dict.get('wind_speed', '-')}m/s, {weather_dict.get('precipitation_prob', '-')}%"
-            weather_cache[region] = (now, weather_dict)
-            return {"weather_data": weather_dict}
-        return {"weather_data": {"temperature": "-", "humidity": "-", "wind_speed": "-", "precipitation_prob": "-"}}
+        if kakao_key:
+            kakao_url = f"https://dapi.kakao.com/v2/local/search/address.json?query={urllib.parse.quote(region)}"
+            headers = {"Authorization": f"KakaoAK {kakao_key}"}
+            k_resp = requests.get(kakao_url, headers=headers, timeout=5)
```

Ruff (0.15.10)

[warning] Async functions should not call blocking HTTP methods

(ASYNC210)